qlib/kernel/asm/aarch64.rs

```diff
@@ -99,7 +99,11 @@ pub fn IRet(kernelRsp: u64) -> ! {
 
 #[inline]
 pub fn GetRsp() -> u64 {
-    return 0;
+    let rsp: u64;
+    unsafe {
+        asm!("mov {0}, sp", out(reg) rsp);
```

Comment:
could you please clarify the usage of GetRsp.
In aarch64, sp can be both userspace and kernel space stack pointer.
If for all instances here you need to have the kernel one, please use sp_el1 otherwise, 
it should be okay.

Reply:
the same goes for functions like `pub fn Rdtsc()` -- arm doesn't have `rdtsc` in the first place! but we are keeping the naming to be consistent with existing code.  To resolve this we need refactoring with another indirection.

Reply:
My comment regards functional correctness, not naming consistency.
If, for instance, in an exception situation you are reading the sp, what would the code be expecting, the kernel stack or the userspace stack. With this consideration, you may need to configure for aarch64 to ensure that, while chasing consistent abstraction, functional correctness is maintained.

Reply:
Yes actually there are many codes that need to be reconstructed, but we are trying to build a runnable Quark on aarh64 so the clean codes is not yet in our consideration and I think we can make some big reconstruction of codes after a runnable Quark is ready.

Reply:
@SchwarzStein @chl337  I have submit another PR to rename the GetRsp to GetCurrentKernelSp and implement it in aarch64, although we still have to get sp_el1 by `mov {0} sp`, we set the SPSel to 1 before we do it. #1008 

And the reason we don't use `mrs` is that we can only mrs sp_el1 in EL2 and EL3 according to this webpage https://forums.raspberrypi.com/viewtopic.php?t=226080